### PR TITLE
Add fcm

### PR DESCRIPTION
**Pull request by osamaasifoak**

### CODE REVIEW

File: android/app/google-services-old.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{
```
Comment (michaelcspeed) [inline android/app/google-services-old.json]: This file has sensitive information in it. It should not be committed to a public git repo 😬

**Review by michaelcspeed — CHANGES_REQUESTED** [review]

File: android/app/google-services-old.json (hunk 1 above)
Comment (michaelcspeed) [reply android/app/google-services-old.json]: I just realised one is "old". I guess that means there was one there already.. can we remove the new one from git?

File: lib/services/notifications/notifications_service.dart [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  await initialiazeLocalNotification();
+  var settings = await _messaging.requestPermission(
+    alert: true,
+    badge: true,
+    provisional: false,
+    sound: true,
+  );
+
+  if (settings.authorizationStatus == AuthorizationStatus.authorized) {
+    print('User granted permission');
```
Comment (michaelcspeed) [inline lib/services/notifications/notifications_service.dart]: Don't need this?

File: lib/services/notifications/notifications_service.dart [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    // For handling the received notifications
+    FirebaseMessaging.onMessage.listen((RemoteMessage message) {
+      _showNotification(
+        message.notification?.title,
+        message.notification?.body,
+        json.encode(message.data),
+      );
+    });
+  } else {
+    print('User declined or has not accepted permission');
```
Comment (michaelcspeed) [inline lib/services/notifications/notifications_service.dart]: I don't think we need this either, we will not see it in prod anyway

File: lib/services/notifications/notifications_service.dart [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    iOS: initializationSettingsIOS,
+  );
+  await flutterLocalNotificationsPlugin.initialize(
+    initializationSettings,
+    onDidReceiveNotificationResponse: onSelect,
+  );
+}
+
+void onSelect(NotificationResponse? data) async {
+  print(data);
```
Comment (michaelcspeed) [inline lib/services/notifications/notifications_service.dart:64]: This is a todo I'm guessing?

**Review by michaelcspeed — CHANGES_REQUESTED** [review]

File: android/app/google-services-old.json (hunk 1 above)
Comment (osamaasifoak) [reply android/app/google-services-old.json]: Sure, I can remove it no problem. I noticed it was already present in the develop branch and I initially thought it was uploaded by someone else. However, I will go ahead and remove the new one. ;)

File: lib/services/notifications/notifications_service.dart (hunk 4 above)
Comment (osamaasifoak) [reply lib/services/notifications/notifications_service.dart:64]: Haha yes, I will create another PR for this

**Review by michaelcspeed — APPROVED** [review]

**Pull request merged**